tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.Framework.cs

```diff
@@ -69,23 +78,86 @@ internal SecurityCoordinator(Security security, HttpContext context, Span span)
     internal IDictionary<string, object> GetPathParams() => _context.Request.RequestContext.RouteData.Values.ToDictionary(c => c.Key, c => c.Value);
 
     /// <summary>
-    /// Initializes a new instance of the <see cref="Coordinator.SecurityCoordinator"/> class.
-    /// framework can do it all at once, but framework only unfortunately
+    /// Framework can do it all at once, but framework only unfortunately
     /// </summary>
     internal void CheckAndBlock(Dictionary<string, object> args)
     {
         var result = RunWaf(args);
         if (result.ShouldBeReported)
         {
+            var reporting = MakeReportingFunction(result.Data, result.AggregatedTotalRuntime, result.AggregatedTotalRuntimeWithBindings);
```

Comment:
I worry about the new allocations required to implement this delegate. Is this code path currently exercised in our throughput/benchmark testing so we can measure the impact?

Reply:
We do have a throughput for blocking, but not on this particular code path, as we only currently benchmark core not framework.

I think there is quite a bit of work that could be done around optimizing ASM, but I believe the impact of the allocation required to implement this closure will be pretty significant compared the allocations required to execute the WAF.